courses/static_analysis_suite/150_lab_customization.rst

```diff
@@ -0,0 +1,241 @@
+
+  * In the Message History menu of the CodePeer Report, check only "Removed"
+  * Several warnings should be displayed
+  * Expand the :ada:`Cruise` project to see all files
+
+* The result should show that those removed warnings belonged to :file:`cruise-legacy.adb`
+
+.. image:: codepeer/cruise_exclude_legacy.png
+
+=================
+Change Settings
+=================
+
+-------------------------------
+Settings Via The Project File
+-------------------------------
+
+* We want to set the default value of the CodePeer parameters in the GPR file.
+
+  * The :toolname:`GPRBuild` attribute to set the switches is :ada:`CodePeer.Switches`
+
+* Change the default level of analysis to **2**
+
+  * The corresponding switch is :ada:`--level`
+
+* Set multiprocessing to **0** cores (ie. let the tool chose).
+
+  * The corresponding switch is :ada:`-j`
+
+* Run a new analysis with :menu:`CodePeer` |rightarrow| :menu:`Analyze All`
+
+* The analysis should return fewer results
+
+----------------------------------
+Overriding Project File Settings
+----------------------------------
+
+* Configure a new analysis with :menu:`CodePeer` |rightarrow| :menu:`Analyse...`
+
+* In the opened window, you can modify the command used to call CodePeer directly at the bottom of the screen.
+
+.. image:: codepeer/cruise_analysis_manual_switches.png
+
+* Notice that there is also an history for this bar.
+
+  * Remove the :ada:`--level` switch to use the default level.
+
+===================
+Tools Integration
+===================
+
+---------------
+GNAT Warnings
+---------------
+
+* CodePeer has integration to several tools, namely the GNAT compiler Warnings and GNATcheck.
+* We will use a bit of the GNAT warnings integration there.
+
+* Configure a new analysis with :menu:`CodePeer` |rightarrow| :menu:`Analyse...`
+
+   * The *Warnings ...* choice is actually a **button**, click on it
+
+.. image:: codepeer/cruise_analysis_warnings_button.png
+
+* A new menu appears with a list of warnings to set.
+
+.. image:: codepeer/cruise_analysis_warnings_menu.png
+
+  * Full list of GNAT Warnings can be found in *GNAT User's Guide*
+  * Depending on your version of the compiler and target, some warnings will or will not be available.
+  * Warning are sorted in the order of their switch
+
+    * Eg. "Most optional warnings" :ada:`-gnatwa` is first, then "Failing assertions" :ada:`-gnatw.a`...
+
+---------------------------------
+Configure Which Warnings To See
+---------------------------------
+
+* Each warning can be either unselected, or a bar or a checkmark
+
+  .. image:: codepeer/cruise_analysis_warnings_bar.png
+    :width: 50%
+
+  * Bar - warning is implicitly checked by CodePeer.
+
+  .. image:: codepeer/cruise_analysis_warnings_check.png
+    :width: 50%
+
+  * Checkmark - warning is explicitly checked by CodePeer.
+
+* Enable GNAT Warnings for
+
+  - Variables that could be constant (`-gnatwk`)
```

Comment:
Change "constant" to "constants" (to match the actual wording displayed in the window)

Reply:
Fixed